document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -246,22 +251,23 @@ protected String parseQueryForNonCompositeFilter(Filter filter) {
       case NOT_EXISTS:
         // TODO: Checks if key does not exist
       case NEQ:
-        throw new UnsupportedOperationException("Only Equality predicate is supported");
       default:
         throw new UnsupportedOperationException(
             String.format("Query operation:%s not supported", op));
     }
-    return filterString.append("'").append(value).append("'").toString();
+    String filters = filterString.append(QUESTION_MARK).toString();
+    paramsBuilder.addStringParam(String.valueOf(value));
```

Comment:
Is every param treated as a string? Would that work?
If that is the case, why do we need the other methods like `addLongParam`/`addFloatParam` in Params?

Reply:
Currently all params are treated, as string, have added LongParam for future use cases

Reply:
What was the value here - `filterString.append("'").append(value).append("'").toString()`?
Here value is an `Object` so, it `toString` method used, right?